miden-crypto/src/merkle/smt/large/mod.rs

```diff
@@ -0,0 +1,1356 @@
+//! Large-scale Sparse Merkle Tree backed by pluggable storage.
+//!
+//! `LargeSmt` stores the top of the tree (depths 0–23) in memory and persists the lower
+//! depths (24–64) in storage as fixed-size subtrees. This hybrid layout scales beyond RAM
+//! while keeping common operations fast. With the `rocksdb` feature enabled, the lower
+//! subtrees and leaves are stored in RocksDB. On reopen, the in-memory top is reconstructed
+//! from cached depth-24 subtree roots.
+//!
+//! Examples below require the `rocksdb` feature.
+//!
+//! Open an existing RocksDB-backed tree:
+//! ```no_run
+//! use miden_crypto::merkle::{LargeSmt, RocksDbConfig, RocksDbStorage};
+//!
+//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let storage = RocksDbStorage::open(RocksDbConfig::new("/path/to/db"))?;
+//! let smt = LargeSmt::new(storage)?; // reconstructs in-memory top if data exists
+//! let _root = smt.root()?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! Initialize an empty RocksDB-backed tree and bulk-load entries using mutations:
+//! ```no_run
+//! use miden_crypto::{
+//!     Felt, Word,
+//!     merkle::{LargeSmt, RocksDbConfig, RocksDbStorage},
+//! };
+//!
+//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let path = "/path/to/new-db";
+//! if std::path::Path::new(path).exists() {
+//!     std::fs::remove_dir_all(path)?;
+//! }
+//! std::fs::create_dir_all(path)?;
+//!
+//! let storage = RocksDbStorage::open(RocksDbConfig::new(path))?;
+//! let mut smt = LargeSmt::new(storage)?; // empty tree
+//!
+//! // Prepare initial entries
+//! let entries = vec![
+//!     (
+//!         Word::new([Felt::new(1), Felt::new(0), Felt::new(0), Felt::new(0)]),
+//!         Word::new([Felt::new(10), Felt::new(20), Felt::new(30), Felt::new(40)]),
+//!     ),
+//!     (
+//!         Word::new([Felt::new(2), Felt::new(0), Felt::new(0), Felt::new(0)]),
+//!         Word::new([Felt::new(11), Felt::new(22), Felt::new(33), Felt::new(44)]),
+//!     ),
+//! ];
+//!
+//! // Compute and atomically apply the initial bulk load
+//! let mutations = smt.compute_mutations(entries)?;
+//! smt.apply_mutations(mutations)?;
```

Comment:
It may be convenient to have an API that wraps this and exposes a single function (for when users don't care about first computing mutations and then applying them). Let's create an issue for this.